integration/tests/epochs/suite.go

```diff
@@ -165,11 +161,14 @@ func (s *Suite) StakeNode(ctx context.Context, env templates.Environment, role f
 	require.NoError(s.T(), err)
 	require.NoError(s.T(), result.Error)
 
-	tokenAmount, err := s.client.TokenAmountByRole(role)
-	require.NoError(s.T(), err)
+	// if node has a machine account key encode it
+	var encMachinePubKey []byte
+	if machineAccountKey != nil {
+		encMachinePubKey, err = flow.EncodeRuntimeAccountPublicKey(machineAccountPubKey)
+		require.NoError(s.T(), err)
+	}
 
-	encMachinePubKey, err := flow.EncodeRuntimeAccountPublicKey(machineAccountPubKey)
-	require.NoError(s.T(), err)
+	containerName := fmt.Sprintf("%s_test", role)
```

Comment:
We'll need to change this for any cases where we add >1 new nodes of the same role.